File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/maps/Polygons.java
package org.maplibre.android.maps;


import androidx.annotation.NonNull;

import org.maplibre.android.annotations.Polygon;
import org.maplibre.android.annotations.PolygonOptions;

import java.util.List;

/**
 * Interface that defines convenient methods for working with a {@link Polygon}'s collection.
 */
interface Polygons {
  Polygon addBy(@NonNull PolygonOptions polygonOptions, @NonNull MapLibreMap maplibreMap);

  List<Polygon> addBy(@NonNull List<PolygonOptions> polygonOptionsList, @NonNull MapLibreMap maplibreMap);

  void update(Polygon polygon);

  List<Polygon> obtainAll();
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/Polygons.kt
package org.maplibre.android.maps

import org.maplibre.android.annotations.Polygon
import org.maplibre.android.annotations.PolygonOptions

/**
 * Interface that defines convenient methods for working with a [Polygon]'s collection.
 */
internal interface Polygons {
    fun addBy(
        polygonOptions: PolygonOptions,
        maplibreMap: MapLibreMap,
    ): Polygon

    fun addBy(
        polygonOptionsList: List<PolygonOptions>,
        maplibreMap: MapLibreMap,
    ): List<Polygon>

    fun update(polygon: Polygon)

    fun obtainAll(): List<Polygon>
}
